lib/rules/indent.js

```diff
@@ -966,7 +983,11 @@ module.exports = {
             ConditionalExpression(node) {
                 const tokens = getTokensAndComments(node);
 
-                if (!(node.parent.type === "ConditionalExpression" && options.flatTernaryExpressions)) {
+                if (options.flatTernaryExpressions && !isFirstTokenOfStatement(tokens[0], node)) {
+                    if (node.parent.type !== "ConditionalExpression") {
+                        offsets.increaseOffset(tokens[0], 1);
```

Comment:
The tests still seem to pass if I remove the entire `if (node.parent.type !== "ConditionalExpression")` statement. Is that intended? If so, I think it would be simplest to just remove the statement and replace the handler with

```js
if (!options.flatTernaryExpressions || isFirstTokenOfStatement(tokens[0], node)) {
    offsets.setDesiredOffsets(tokens, tokens[0], 1);
}
```